### ISSUE TITLE
Problem with onboarding Photon in some environments

### ISSUE THREAD

**Issue by Komoi** [particle-iot/spark-setup-android#51]
Hi,

we are developing product based on particle platform and we are using Photon device, which is integrated into our hardware. We have problems with onboarding using Android devices, iOS ones are mostly fine. This onboarding problem happens in approx 80% of tries when in specific environment so it actually succeeds sometimes. This problem happen only on specific wifi networks and can not be reproduced everywhere. Result is that process fails on step 5 - checking if device was claimed. Problem probably occurs earlier - if we watch successful onboarding in Particle console, we can see that device comes online around step 2 in newly added graphical event log. In case that onboarding fails, we can not see any activity in Particle console.

During the onboarding we can se various error logs in logcat. Some of them are "normal" for us and we know, that even if they happen, onboarding can succeed. One stacktrace, which is repeating during step 5 when device fails to set up is this:

` I/CheckIfDeviceClaimedStep: Running step CheckIfDeviceClaimedStep
 E/ParticleCloudException: best msg: Permission Denied
 W/System.err: io.particle.android.sdk.cloud.ParticleCloudException
 W/System.err:     at io.particle.android.sdk.cloud.ParticleCloud.getDevice(ParticleCloud.java:525)
 W/System.err:     at io.particle.android.sdk.cloud.ParticleCloud.getDevice(ParticleCloud.java:316)
 W/System.err:     at io.particle.android.sdk.devicesetup.setupsteps.CheckIfDeviceClaimedStep.onRunStep(CheckIfDeviceClaimedStep.java:73)
 W/System.err:     at io.particle.android.sdk.devicesetup.setupsteps.SetupStep.runStep(SetupStep.java:54)
 W/System.err:     at io.particle.android.sdk.devicesetup.setupsteps.SetupStepsRunnerTask.runSteps(SetupStepsRunnerTask.java:76)
 W/System.err:     at io.particle.android.sdk.devicesetup.setupsteps.SetupStepsRunnerTask.doInBackground(SetupStepsRunnerTask.java:40)
 W/System.err:     at io.particle.android.sdk.devicesetup.setupsteps.SetupStepsRunnerTask.doInBackground(SetupStepsRunnerTask.java:13)
 W/System.err:     at android.os.AsyncTask$2.call(AsyncTask.java:305)
 W/System.err:     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
 W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
 W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
 W/System.err:     at java.lang.Thread.run(Thread.java:761)`

We wonder what might cause this inconsistency (that onboarding is sometimes working and sometimes not). It would be best if you would be open to call to discuss this problem. Let me know, if you want to see full logs form onboarding.

Thanks,
Ondrej

**Comment by CityVibes** [comment]
Hi @Komoi,
Which version of device setup library are you using?

**Comment by Komoi** [comment]
Hi, I am using version 0.4.3. We have used static copy of library code because we needed to customise UI quite a lot. Logic is basically untouched.

**Comment by CityVibes** [comment]
@Komoi Well you are using older version, latest setup library version is 0.4.9, although issue is likely still there.
You mentioned "specific environment" and "specific wifi networks", could you tell me more about wifi connections a bit more, do they have anything in common, firewalls, any kind of configuration. Trying to reproduce bug so full logs would be useful as well.

**Comment by towynlin** [comment]
Also, the "permission denied" part of the error is probably coming from the Particle API, meaning the device may be owned by another user, may not be a part of the requested product, or another similar permission error.

**Comment by Komoi** [comment]
We have trouble to reproduce the issue consistently even on mentioned specific wifi networks. We are not even sure if this lead is correct. We think that this might be caused because wifi is dualband and phone is connected to 5GHz network before pairing. I am not sure about firewall or other configuration, because I do not have physical access to that network. You can see onboarding logs here (but some of them may be our custom ones)

Permission denied would mean that device was no unclaimed correctly, but it should be part of claiming process. We also experienced increased fail rate when onboarding right after unclaiming.

https://pastebin.com/riS9mAPA
https://pastebin.com/NJg1Hqnu

**Comment by Komoi** [comment]
Also, now I experienced problem with unclaiming during the onboarding process. If I try to claim already claimed device with same account it belongs to, it works, because claim is checked by listing devices from you account and checking, if one you are onboarding is there. When I try to do this with another account (one which does not own the device) it fails every time. This is probably different issue than one I was referring to originally, I just found it worth mentioning.
